### ISSUE TITLE
Issue when implementing graceful shutdown

### ISSUE THREAD

**Issue by JoHuang** [JaniAnttonen/winston-loki#26]
I'm implementing graceful shutdown of my app, but found an issue might be related to this project.

### Minimal Reproducible Code
```js
"use strict";
require("dotenv").config();

const winston = require("winston");
const { combine, timestamp, label, printf, colorize } = winston.format;
const LokiTransport = require("winston-loki");
const env = process.env.NODE_ENV || "development";
const level = env === "development" ? "debug" : "info";
const name = "express";

const Log = winston.createLogger({
    level : level,
    transports: [
        new winston.transports.Console({
            format: combine(
                label({ label: name }),
                colorize(),
                timestamp(),
                printf(({ level, message, label, timestamp }) => {
                    return `${timestamp} ${level}: [${label}] ${typeof message == "string" ? message : JSON.stringify(message)}`;
                }),
            ),        
        }),
        // works if commented
        new LokiTransport({
            host: process.env.GRAFANA_LOKI_URL
        }),
    ]
});
const express = require("express");
const app = express();
const server = require("http").Server(app);

server.listen(8080, () => {
    Log.info("The server is running...");
});

const gracefullyShutdown = () => {
    Log.info(`SIGTERM / SIGINT signal received.`);
    Log.info('Closing http server.');
    server.close(() => {
        Log.info('Http server closed.');

        setTimeout(()=> {
            Log.info('Shutdown after 5 secs!');
            
        }, 5000);
    });
};

process.on('SIGTERM', gracefullyShutdown);
process.on('SIGINT' , gracefullyShutdown);

process.once('exit', () => {
    console.trace("I am here");
})

process.on('unhandledRejection', (reason, p) => {
    console.log('Unhandled Rejection at:', p, 'reason:', reason);
});

```

### Explanation

1. if I add `LokiTransport` to winston transport as my code shown, when I press Ctrl + C to trigger SIGINT, the app exit immediately without waiting. here is the log:
```
2020-02-12T13:20:25.440Z info: [express] The server is running...
^C2020-02-12T13:20:27.553Z info: [express] SIGTERM / SIGINT signal received.
2020-02-12T13:20:27.553Z info: [express] Closing http server.
2020-02-12T13:20:27.553Z info: [express] Http server closed.
Trace: I am here
    at process.<anonymous> (/Users/user/abcproject/index.js:54:13)
    at Object.onceWrapper (events.js:300:26)
    at process.emit (events.js:215:7)
    at exit (internal/process/per_thread.js:158:15)
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
```

2. If I commented `LokiTransport`,  when I press Ctrl + C to trigger SIGINT, the app exit after 5s as my expected. here is the log:
```
2020-02-12T13:18:51.733Z info: [express] The server is running...
^C2020-02-12T13:18:53.706Z info: [express] SIGTERM / SIGINT signal received.
2020-02-12T13:18:53.707Z info: [express] Closing http server.
2020-02-12T13:18:53.707Z info: [express] Http server closed.
2020-02-12T13:18:58.713Z info: [express] Shutdown after 5 secs!
Trace: I am here
    at process.<anonymous> (/Users/user/abcproject/index.js:68:13)
    at Object.onceWrapper (events.js:300:26)
    at process.emit (events.js:210:5)
```

Is there any possible reason to explain this behavior? Maybe because of some dependency library?
Or any way to find out what code causes early exit?

Thanks

**Comment by JaniAnttonen** [comment]
Thanks again for your contribution! I might've done a whoopsie with this. As of my latest knowledge, this library may now shut down gracefully, but only by itself. This means that it only waits for its own internal state (any unsent batches or unfinished requests) to finish. I don't have a fast answer for this but to make a patch to disable graceful shutdown with a configuration flag.

**Comment by JaniAnttonen** [comment]
And you're right, there's a dependency that is used to handle the graceful shutdown. It's called `async-exit-hook`

**Comment by JaniAnttonen** [comment]
Then again, it only does this:
```javascript
exitHook(callback => {
      this.sendBatchToLoki()
        .then(() => callback())
        .catch(() => callback())
    })
```
That __shouldn't__ trigger exit by itself on first glance, since the promised behavior is "Run some code when the process exits" but there's a part in `async-exit-hook` that does something with `process.exit` that might be the culprit:
```javascript
function doExit() {
	if (doExitDone) {
		return;
	}
	doExitDone = true;

	if (exit === true) {
		// All handlers should be called even if the exit-hook handler was registered first
		process.nextTick(process.exit.bind(null, code));
	}
}
```

**Comment by JaniAnttonen** [comment]
@JoHuang, I published a new version, `5.1.1` on npm that has a hotfix mentioned here: https://github.com/JaniAnttonen/winston-loki/releases/tag/v5.1.1

**Comment by JoHuang** [comment]
@JaniAnttonen I added close method for manually send the remaining log data after receiving signals.
Now you can call `logger.close()` or `logger.end()` to trigger LokiTransport's close method.

```js
const gracefullyShutdown = () => {
    Log.info(`SIGTERM / SIGINT signal received.`);
    Log.info('Closing http server.');
    server.close(() => {
        Log.info('The final log.');
        Log.close();
        // wait for few secs to let transport send data to remote server
        setTimeout(()=> {
            process.exist(0);
        }, 5000);
    });
};
```
